### PR TITLE
[fix](schemachange) fix the schema change that causes the be core dump.

### LINKED ISSUE #14803: [Bug] schema change causes be core dump
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

maste

### What's Wrong?

daily: https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/lastCompletedBuild/testReport/(root)/test_sys_alter_schema_change_modify/test_modify_varchar_to_number/

```
sql: ALTER TABLE s11a8_ema_change_modify_test_modify_varchar_to_number_db.s10ae_ema_change_modify_test_modify_varchar_to_number_tb2 MODIFY COLUMN k8 FLOAT, MODIFY COLUMN k9 DOUBLE;

#0  doris::KeyCoder::full_encode_ascending (this=0x0, value=0x7f5569e3fcd0, buf=0x7f5389fdf4d8) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/key_coder.h:49
#1  0x000000000a632cfa in doris::segment_v2::SegmentWriter::_full_encode_keys[abi:cxx11](std::vector<doris::vectorized::IOlapColumnDataAccessor*, std::allocator<doris::vectorized::IOlapColumnDataAccessor*> > const&, unsigned long, bool) (this=0x7f5569e66600, key_columns=..., pos=0, null_first=true) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/rowset/segment_v2/segment_writer.cpp:264
#2  0x000000000a632677 in doris::segment_v2::SegmentWriter::append_block (this=0x7f5569e66600, block=0x7f5569e63180, row_pos=0, num_rows=1)
    at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/rowset/segment_v2/segment_writer.cpp:212
#3  0x000000000aa41798 in doris::BetaRowsetWriter::_add_block (this=0x7f5569e59000, block=0x7f5569e63180, segment_writer=0x7f5569e59128)
    at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/rowset/beta_rowset_writer.cpp:573
#4  0x000000000aa413ee in doris::BetaRowsetWriter::add_block (this=0x7f5569e59000, block=0x7f5569e63180) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/rowset/beta_rowset_writer.cpp:120
#5  0x000000000a07fd3d in doris::VSchemaChangeDirectly::_inner_process (this=0x7f5569ec1de0, rowset_reader=..., rowset_writer=0x7f5569e59000, new_tablet=..., base_tablet_schema=...)
    at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/schema_change.cpp:1343
#6  0x000000000a09a6c6 in doris::SchemaChange::process (this=0x7f5569ec1de0, rowset_reader=..., rowset_writer=0x7f5569e59000, new_tablet=..., base_tablet_schema=...)
    at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/schema_change.h:105
#7  0x000000000a089215 in doris::SchemaChangeHandler::_convert_historical_rowsets (sc_params=...) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/schema_change.cpp:2201
#8  0x000000000a086bb9 in doris::SchemaChangeHandler::_do_process_alter_tablet_v2 (request=...) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/schema_change.cpp:2008
#9  0x000000000a084672 in doris::SchemaChangeHandler::process_alter_tablet_v2 (request=...) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/olap/schema_change.cpp:1757
```

### What You Expected?

no core dump

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
Forbid schema change to add or modify the key column of the agg model as double or float.

# Proposed changes

Issue Number: close #14803

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.82 seconds
 load time: 478 seconds
 storage size: 17123356247 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221204135443_clickbench_pr_57499.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AlterTableStmt.java [hunk 1]
```diff
@@ -77,7 +78,15 @@ public void analyze(Analyzer analyzer) throws UserException {
...
             ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_ALTER_OPERATION);
         }
+        Table table = null;
+        if (!FeConstants.runningUnitTest) {
+            table = Env.getCurrentInternalCatalog().getDbOrDdlException(tbl.getDb())
+                    .getTableOrDdlException(tbl.getTbl());
+        }
         for (AlterClause op : ops) {
+            if (op instanceof AlterTableClause && table instanceof OlapTable) {
+                ((AlterTableClause) op).setOlapTable((OlapTable) table);
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AlterTableStmt.java:88]: it better to table name no OlapTable object

Comment (luozenglin) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/AlterTableStmt.java]: Done

**Review by morningman — APPROVED** [review]
LGTM

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**